frontend/app/src/components/history/events/HistoryEventsAction.vue

```diff
@@ -104,11 +110,23 @@ function hideAddAction(item: HistoryEvent): boolean {
           </template>
           {{ event.ignoredInAccounting ? t('transactions.unignore') : t('transactions.ignore') }}
         </RuiButton>
-        <template v-if="evmEvent">
+        <template v-if="blockEvent">
+          <RuiButton
+            variant="list"
+            :disabled="loading || eventTaskLoading"
+            @click="redecode({ type: blockEvent.entryType, data: [blockEvent.blockNumber] })"
```

Comment:
wouldn't it make more sense for this and the one below to pass the blockeven/evmEvent to the method and do the transformation to the `PullEventPayload` in the function in the code?